lib/rake_helper.rb

```diff
@@ -28,12 +28,23 @@ def run_sql(sql, action: :execute)
   end
 
   private
+  def obtain_message(message, *params)
+    params.empty? ? message : params_to_message(params)
+  end
 
   def put_and_log(message:, type:)
     puts "#{Time.now} #{message}"
     Rails.logger.public_send(type, message)
   end
 
+  def params_to_message(params)
+    message_from_hash(params.last)
+  end
+
+  def message_from_hash(sql_hash)
+    sql_hash.is_a?(Hash) && sql_hash[:sql] ? sql_hash[:sql] : 'Unidentified Call'
+  end
+
   def method_missing(method, *args)
     super unless Logger::Severity.constants.include?(method.upcase)
```

Comment:
Just a though: we could have also add the state methods `[:start, :finish, :failure]` on `method_missing` as exceptions,  for it calls `put_and_log` as well; however it could be involved to the params handling.